### PR TITLE
feat(ann2snn): add STA transformer conversion

### PR DESCRIPTION
**Pull request by AllenYolk**
## Summary
- add `STATransformerRecipe` for online STA-style Transformer ANN2SNN conversion
- expose the recipe through ann2snn recipe exports and update converter return typing for recipe finalization
- add focused tests for STA validation, reset behavior, MHA paths, constants, dict outputs, threshold modes, and calibration limits

## Scope
- This PR intentionally keeps STA as an online/stateful implementation. Layer-wise full-sequence TD execution is deferred to Phase 5B.
- This PR intentionally does not add tutorial content; the Transformer conversion tutorial should be separate from the rate-coding tutorial.

## Validation
- `uv format --check -- spikingjelly/activation_based/ann2snn/recipes/sta_transformer.py spikingjelly/activation_based/ann2snn/recipes/base.py spikingjelly/activation_based/ann2snn/converter.py test/activation_based/test_ann2snn_transformer.py test/activation_based/test_ann2snn.py`
- `uv run pytest -q test/activation_based/test_ann2snn.py test/activation_based/test_ann2snn_transformer.py` -> 218 passed
- `cd docs && make html` -> build succeeded
- g2 ImageNet val, ViT-B/16 baseline top1/top5: 0.81068 / 0.95318
- g2 ImageNet val, STA spiking_encoder T=8 threshold_scale=0.5 top1/top5: 0.80590 / 0.95074; drop: 0.00478

## Summary by CodeRabbit

* **New Features**
  * Added an STA Transformer ANN-to-SNN conversion recipe with optional calibration-based thresholding and configurable `time_steps`.
  * Re-exported `STATransformerRecipe` as part of the public `ann2snn` API.
* **Bug Fixes**
  * Improved handling of the `"sta_transformer"` recipe alias with clearer guidance for correct usage.
* **Documentation**
  * Updated conversion/recipe return type hints to reflect returning a `torch.nn.Module`.
* **Tests**
  * Added end-to-end STA Transformer conversion and calibration tests, covering attention/masking variants, determinism, and validation errors.

### CODE REVIEW

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist please review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in 126bd77ccbe8bbf84b98fdd5dfea44c5aa5304f4. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in 1d88fbe83049cd65254811f95772c89930f1a6fb. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in 7975c18e38d6a370476896e93d7459fb4687884f. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in ad1ee1473a597955f1084f29904776732c229645. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in c1f7e5bc4d951d16a0c72149853b7f84c94dc89c. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in 1f318b6cc1ded6adbc80b6064f4e867ad12b52c9. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in 9d48491affa0f8103e6a6d6245190bd05889e17e. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in 5c744caa67cc4883c5923756531e6bdb60399939. Please re-review the latest PR state.

**Comment by AllenYolk** [comment]
@coderabbitai @gemini-code-assist review fixes pushed in d14ca37492704ec4079108799c94e25a4bfccc92. Please re-review the latest PR state.
